### ISSUE TITLE
Day 21 Planner device auth works with admin user, fails with end user

### ISSUE THREAD

**Issue by skillriver** [microsoftgraph/dotnetcore-console-sample#40]
The code for day 21 works fine when I do device code auth as an admin user, but fails when I try to do it with a normal end user. In the console app I get: Unhandled Exception: Microsoft.Identity.Client.MsalUiRequiredException: AADSTS50097: Device authentication is required.

It is not clear to me if this is supposed to be able to run as a delegated permission using just normal work accounts in my organization?

**Comment by BrianTJackett** [comment]
Thanks for reporting.  Will take a look at this when get a chance.  I have only tested with an admin account as well thus far.

**Comment by BrianTJackett** [comment]
@skillriver I was able to successfully run the sample as a non-admin user.  Can you confirm the following:

- Using an Azure AD app that has allowed public client in the manifest?
- Add API permissions to Azure AD app for Groups.ReadWrite.All and User.ReadBasic.All delegated permissions?
- Azure AD app has been provided with admin consent?
- When you are running the console application you see the "To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code <codeRemoved> to authenticate." prompt in your command line output?

**Comment by skillriver** [comment]
Yes, can confirm all of the above. And after devicelogin is successfully entered (4. above), I'm presented with the work account login (5.), login and gets success in the browser for devicelogin completed (6.). It's after this, in the console output, I get the error provided. (7.)

Btw, should I get a user consent dialog (for the delegated perms), sometime between 5. and 6.? I don't but wondered if it should be..?

**Comment by gavinbarron** [comment]
The consent for delegated permissions is not shown as the admin consented when configuring the AAD app. It's not super clear but it does say during this process that you are granting consent for all users in your tenant.

**Comment by skillriver** [comment]
Thanks @gavinbarron, thought so as well, but just double checking.

**Comment by BrianTJackett** [comment]
@skillriver Do you have any conditional access policies, multi-factor authentication, or similar policies in place for the account that you are logging into?  Sounds like others have run into similar issue in the past.

https://stackoverflow.com/questions/39364309/authentication-from-mobile-devices-with-openid-connect-azure-ad-not-working-wi#comment66236359_39364309

**Comment by skillriver** [comment]
@BrianTJackett, you're the man! ;) I had been thinking about it, even the other day I checked the sign-in logs both in the AAD portal and via Log Analytics, and couldn't find any Conditional Access errors. But, I had implemented some Cloud App Security Proxy policies for SharePoint Online and Exchange Online, these applies to browser and monitor sessions after authenticating. So when I removed my end user from those CA's, everything worked correctly :)
